apps/schools/domains/employee/components/currentEmployee/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from 'react'
+import { Col, Row, Typography } from 'antd'
+import Image from 'next/image'
+
+import { getUuidFromUrl } from '@domains/common/utils/getUuidFromUrl'
+
+import styles from './styles/styles.module.scss'
+import { ErrorType } from '@store/commonApi'
+import router from 'next/router'
+import duckEmptyPage from '@public/image/duckEmptyPage.svg'
+import { Button } from '@domains/common/components/button'
+import { Field } from '@domains/common/components/field'
+import { ActionBar } from '@domains/common/components/stickyBlock/actionBar'
+import DeleteModal from '@domains/common/components/deleteModal'
+import { useDeleteEmployeeByIdMutation, useGetEmployeeQuery } from '@domains/employee/redux/employeeApi'
+
+const CurrentEmployee = () => {
+    const [isModalVisible, setIsModalVisible] = useState(false)
+    const [mutation, isDeleteFinished] = useDeleteEmployeeByIdMutation()
+    const uuid = getUuidFromUrl()
+
+    const {
+        data: employee,
+        error: employeeError,
+    } = useGetEmployeeQuery({
+        employee_id: uuid[0],
+    })
+
+    useEffect(() => {
+        if (employeeError && (employeeError as ErrorType).status == 404) {
+            router.push('/employee')
+        }
+    }, [employeeError])
+
+    if (isDeleteFinished.isSuccess) return null
+    if (uuid.length === 0) router.push('/404')
```

Comment:
this is the second time I see such a code, is it possible to make a hook of some kind?

Reply:
yes it will be good to summarize this logic in special function/hook

Reply:
I created [issue in jira](https://lamart.atlassian.net/browse/SCHOOL-790) for that